src/storable.rs

```diff
@@ -492,6 +492,120 @@ where
     };
 }
 
+fn get_size_len<T>() -> usize
+where
+    T: Storable,
+{
+    match T::BOUND {
+        Bound::Bounded {
+            max_size,
+            is_fixed_size,
+        } => bytes_to_store_size(&Bounds {
+            max_size,
+            is_fixed_size,
+        }) as usize,
+        Bound::Unbounded => 8,
+    }
+}
+
+// Serializes `entry` into `bytes` starting from the index `start`
+// in `bytes`.
+// When serialized `entry` is saved in `bytes` on indices `[start, end)`
+// the function will return index `end` - the first index after
+// `start` that is not occupied with the serialization of `entry`.
+fn serialize_with_size<T>(entry_bytes: &[u8], bytes: &mut [u8], start: usize) -> usize
+where
+    T: Storable,
+{
+    let size_len = get_size_len::<T>();
+    let actual_size = entry_bytes.len();
+
+    encode_size_universal::<T>(&mut bytes[start..start + size_len], actual_size);
+
+    bytes[start + size_len..start + size_len + actual_size].copy_from_slice(entry_bytes);
+
+    start + actual_size + size_len
+}
+
+// Deserialize the struct starting at index `start` in `bytes`.
+// When serialized struct is saved in `bytes` on indices `[start, end)` the
+// function will return deserialized struct and index `end` - the first index
+// after `start` that is not occupied with the serialization of the struct.
+fn deserialize_bounded_with_size<T>(bytes: &[u8], start: usize) -> (T, usize)
```

Comment:
This method's name is also misleading - it isn't specific to bounded types anymore.